syntax = "proto3";

package calc.v1;

service Calc {
  rpc Add(AddRequest) returns (AddResponse) {}

  rpc Sum(stream SumRequest) returns (SumResponse) {}

  rpc Divisors(DivisorsRequest) returns (stream DivisorsResponse) {}
}

message AddRequest {
  int32 x = 1;
  int32 y = 2;
}

message AddResponse {
  int32 result = 1;
}

message SumRequest {
  int32 x = 1;
}

message SumResponse {
  int32 result = 1;
}

message DivisorsRequest {
  int32 x = 1;
}

message DivisorsResponse {
  int32 result = 1;
}
